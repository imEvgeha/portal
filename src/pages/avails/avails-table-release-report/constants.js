export const CREATE_REPORT = 'Create Report';
export const NEW_RELEASE_REPORT = 'Report';
export const START_YEAR = 1970;
export const END_YEAR = 2030;
export const MOCK_YEAR = 2099;
export const ERROR_MESSAGE = 'No records found';
export const MONTHS = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
];
