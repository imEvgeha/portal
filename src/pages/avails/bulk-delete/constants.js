export const HEADER = {
    cells: [
        {
            key: 1,
            content: 'Linked Rights',
            width: 30,
            shouldTruncate: true,
        },
        {
            key: 2,
            content: 'Right ID',
            width: 15,
        },
        {
            key: 3,
            content: 'Type',
            width: 10,
        },
        {
            key: 4,
            content: 'Original Right',
            width: 25,
        },
        {
            key: 5,
            content: 'Source Right',
            width: 20,
        },
    ],
};

export const BULK_DELETE_WARNING_MSG = num => `Permanently delete ${num} Rights?`;
export const BULK_DELETE_REMAINING_MSG = (num1, num2) =>
    `${num1} Rights have been deleted. Deleting some of the remaining ${num2} Rights will also delete their linked TPR/Bonus Rights. Please review.`;
export const BULK_DELETE_LINKED_RIGHT_MSG = 'The following linked TPR/Bonus rights will also be marked as deleted:';
export const SELECTED_RIGHTS = 'Selected Rights';
export const AFFECTED_RIGHTS = 'Affected Rights';
export const BULK_DELETE_CONTINUE_MSG = 'How would you like to continue?';
export const BULK_DELETE_BTN_DELETE = 'Mark as Deleted';
export const BULK_DELETE_BTN_CANCEL = 'Cancel';

export const RIGHT_ID = 'Right ID';
export const TITLE = 'Title';
export const LINKED_RIGHT_ID = 'Linked Right ID';
export const TPR_ORIGINAL_RIGHTS = 'TPR Original Rights';
export const BONUS_SOURCE_RIGHT = 'Bonus Source Right';
