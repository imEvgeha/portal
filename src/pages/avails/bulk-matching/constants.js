export const TITLE_MATCHING_MSG = 'Select titles from the repositories that match the incoming right or declare a ';
export const TITLE_MATCHING_REVIEW_HEADER = 'Title Matching Review';
export const RIGHT_TABS = {
    SELECTED: 'Selected Rights',
    AFFECTED: 'Affected Rights',
    BONUS_RIGHTS: 'Existing Bonus Rights',
};
