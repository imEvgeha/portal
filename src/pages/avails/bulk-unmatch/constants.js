export const BULK_UNMATCH_TITLE = 'Bulk Unmatch Rights Review';
export const BULK_UNMATCH_WARNING = 'Warning: Please be aware you need to validate title fields.';
export const BULK_UNMATCH_CONFIRM_BTN = 'Confirm';
export const BULK_UNMATCH_CANCEL_BTN = 'Cancel';

