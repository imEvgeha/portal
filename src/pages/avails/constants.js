export default {
    PAGE_SIZE: 100,
    sortParams: [{id: 'received', desc: 'true'}],
    AVAIL_HISTORY_ID: 'availHistoryIds',
    INGEST_HISTORY_ATTACHMENT_ID: 'ingestHistoryAttachmentId',
    INITIAL_SORT: {colId: 'updatedAt', sort: 'desc'},
};
