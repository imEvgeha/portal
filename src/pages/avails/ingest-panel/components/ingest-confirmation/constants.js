export const INGEST_ASSIGN_MESSAGE = 'You have assigned your upload with:';
export const INGEST_UPLOAD_CONTINUE_MSG = 'Do you wish to continue?';
export const INGEST_UPLOAD_CANCEL_BTN = 'Cancel';
export const INGEST_UPLOAD_CONTINUE_BTN = 'Continue';
