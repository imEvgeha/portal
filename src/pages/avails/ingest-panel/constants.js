export default {
    STATUS_LIST: [
        {value: '', label: 'ALL'},
        {value: 'PENDING', label: 'PENDING'},
        {value: 'MANUAL', label: 'MANUAL'},
        {value: 'COMPLETED', label: 'COMPLETED'},
        {value: 'FAILED', label: 'FAILED'},
    ],
    DATEPICKER_LABELS: ['From', 'To'],
    REPORT: {
        total: {label: 'Rights', value: ''},
        created: {label: 'New', value: 'Created'},
        updated: {label: 'Updated', value: 'Updated'},
        fatal: {label: 'Fatals', value: 'Fatal'},
        pending: {label: 'Pending', value: 'Pending'},
        errors: {label: 'Errors', value: 'Error'},
    },
    ingestTypes: {
        EMAIL: 'Email',
        UPLOAD: 'Upload',
    },
    attachmentTypes: {
        EXCEL: 'Excel',
        EMAIL: 'Email',
    },
    filterKeys: {
        RECEIVED_FROM: 'receivedFrom',
        RECEIVED_TO: 'receivedTo',
        STATUS: 'status',
        LICENSOR: 'licensor',
    },
    URLFilterKeys: {
        receivedFrom: 'igReceivedFrom',
        receivedTo: 'igReceivedTo',
        status: 'igStatus',
        licensor: 'igLicensor',
    },
    SERVICE_REGIONS: [
        {label: 'US', value: 'US'},
        {label: 'UK', value: 'UK'},
    ],
    TEMPLATES: {
        STUDIO: 'Studio',
        USMASTER: 'US Master',
        INTERNATIONAL: 'International',
    },
    LICENSEE_WARNING: 'Any mismatches between the file and the selected licensees will result in a fatal',
};

export const DEBOUNCE_TIMEOUT = 2000;
