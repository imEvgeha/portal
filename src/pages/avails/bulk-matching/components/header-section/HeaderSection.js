import React from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import classNames from 'classnames';
import {RIGHT_TABS} from '../../constants';

const HeaderSection = ({
    activeTab,
    changeActiveTab,
    selectedRights,
    isBonusRight,
    existingBonusRights,
    affectedRights,
    showModal,
    isNewTitleDisabled,
}) => {
    return (
        <div className="nexus-c-bulk-matching__header">
            <div
                className={classNames(
                    'nexus-c-bulk-matching__rights-tab',
                    activeTab === RIGHT_TABS.SELECTED && 'nexus-c-bulk-matching__rights-tab--active'
                )}
                onClick={() => changeActiveTab(RIGHT_TABS.SELECTED)}
            >
                {RIGHT_TABS.SELECTED} ({selectedRights})
            </div>
            {isBonusRight ? (
                <div
                    className={classNames(
                        'nexus-c-bulk-matching__rights-tab',
                        activeTab === RIGHT_TABS.BONUS_RIGHTS && 'nexus-c-bulk-matching__rights-tab--active'
                    )}
                    onClick={() => changeActiveTab(RIGHT_TABS.BONUS_RIGHTS)}
                >
                    {RIGHT_TABS.BONUS_RIGHTS} ({existingBonusRights})
                </div>
            ) : (
                <div
                    className={classNames(
                        'nexus-c-bulk-matching__rights-tab',
                        activeTab === RIGHT_TABS.AFFECTED && 'nexus-c-bulk-matching__rights-tab--active'
                    )}
                    onClick={() => changeActiveTab(RIGHT_TABS.AFFECTED)}
                >
                    {RIGHT_TABS.AFFECTED} ({affectedRights})
                </div>
            )}
            <Button className="nexus-c-bulk-matching__btn" onClick={showModal} isDisabled={isNewTitleDisabled}>
                New Title
            </Button>
        </div>
    );
};

HeaderSection.propTypes = {
    activeTab: PropTypes.string,
    affectedRights: PropTypes.number,
    changeActiveTab: PropTypes.func.isRequired,
    existingBonusRights: PropTypes.number,
    isBonusRight: PropTypes.bool,
    isNewTitleDisabled: PropTypes.bool,
    selectedRights: PropTypes.number,
    showModal: PropTypes.func.isRequired,
};

HeaderSection.defaultProps = {
    activeTab: '',
    affectedRights: 0,
    existingBonusRights: 0,
    isBonusRight: false,
    isNewTitleDisabled: true,
    selectedRights: 0,
};

export default HeaderSection;
