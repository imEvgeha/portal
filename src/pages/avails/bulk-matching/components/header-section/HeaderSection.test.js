import React from 'react';
import Button from '@atlaskit/button';
import {shallow} from 'enzyme';
import {RIGHT_TABS} from '../../constants';
import HeaderSection from './HeaderSection';

describe('HeaderSection', () => {
    let wrapper = null;
    const props = {
        activeTab: RIGHT_TABS.SELECTED,
        affectedRights: 2,
        existingBonusRights: 1,
        isBonusRight: false,
        isNewTitleDisabled: true,
        selectedRights: 1,
        changeActiveTab: () => null,
        showModal: () => null,
    };
    beforeAll(() => {
        wrapper = shallow(<HeaderSection {...props} />);
        wrapper.setProps({
            changeActiveTab: tab => wrapper.setProps({activeTab: tab}),
        });
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a New Title button', () => {
        // eslint-disable-next-line no-magic-numbers
        expect(wrapper.find(Button).length).toEqual(1);
    });

    it('should display selected rights tab on load', () => {
        const selectedTab = wrapper.find('.nexus-c-bulk-matching__rights-tab--active');
        expect(selectedTab.text().includes(RIGHT_TABS.SELECTED)).toBe(true);
    });

    it('should not display affected tab on load', () => {
        const selectedTab = wrapper.find('.nexus-c-bulk-matching__rights-tab--active');
        expect(selectedTab.text().includes(RIGHT_TABS.AFFECTED)).toBe(false);
    });

    it('should not display bonus rights tab on load', () => {
        const selectedTab = wrapper.find('.nexus-c-bulk-matching__rights-tab--active');
        expect(selectedTab.text().includes(RIGHT_TABS.BONUS_RIGHTS)).toBe(false);
    });

    it('should display affected tab on click', () => {
        wrapper.find('.nexus-c-bulk-matching__rights-tab').at(1).props().onClick();
        expect(wrapper.find('.nexus-c-bulk-matching__rights-tab--active').text().includes(RIGHT_TABS.AFFECTED)).toBe(
            true
        );
    });

    it('should not display affected tab when bonus right', () => {
        wrapper.setProps({isBonusRight: true});
        expect(wrapper.find('.nexus-c-bulk-matching__rights-tab').length).toEqual(2);
    });

    it('should display bonus rights tab on click', () => {
        wrapper.setProps({isBonusRight: true});
        wrapper.find('.nexus-c-bulk-matching__rights-tab').at(1).props().onClick();
        expect(
            wrapper.find('.nexus-c-bulk-matching__rights-tab--active').text().includes(RIGHT_TABS.BONUS_RIGHTS)
        ).toBe(true);
    });

    it('should have a new title button and a new title link', () => {
        expect(wrapper.find('.nexus-c-bulk-matching__btn').length).toEqual(1);
    });
});
