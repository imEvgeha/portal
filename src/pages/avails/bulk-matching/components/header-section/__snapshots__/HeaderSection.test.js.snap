// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection should match snapshot 1`] = `
<div
  className="nexus-c-bulk-matching__header"
>
  <div
    className="nexus-c-bulk-matching__rights-tab nexus-c-bulk-matching__rights-tab--active"
    onClick={[Function]}
  >
    Selected Rights
     (
    1
    )
  </div>
  <div
    className="nexus-c-bulk-matching__rights-tab"
    onClick={[Function]}
  >
    Affected Rights
     (
    2
    )
  </div>
  <Button
    className="nexus-c-bulk-matching__btn"
    isDisabled={true}
    onClick={[Function]}
  >
    New Title
  </Button>
</div>
`;
